molecule/driver/openstackdriver.py

```diff
@@ -134,15 +134,23 @@ def up(self, no_provision=True):
                     security_groups=instance.get('security_groups', []))
                 self._reset_known_host_key(server['interface_ip'])
                 instance['created'] = True
+                instance['reachable'] = False
                 num_retries = 0
-                while not self._check_ssh_availability(
-                        server['interface_ip'],
-                        instance['sshuser'],
-                        timeout=6,
-                        sshkey_filename=self._get_keyfile(
-                        )) or num_retries == 5:
+                while num_retries < 5:
                     util.print_info('\t Waiting for ssh availability...')
                     num_retries += 1
+                    if self._check_ssh_availability(
+                            server['interface_ip'],
+                            instance['sshuser'],
+                            timeout=6,
+                            sshkey_filename=self._get_keyfile()):
+                        instance['reachable'] = True
+                        break
+                if not instance['reachable']:
```

Comment:
Just to give some context, the original while loop never matches, because `not ssh_availability` evaluates to `True` when the instance is not reachable, thus the `num_retries == 5` never gets checked.

Since it is possible the instance gets reached on the last iteration, a check on `num_retries` is not enough here.